Fix #653

magicDGS opened the pull request:
Simple patch as suggested in #653. `TribbleIndexedFeatureReader.isGZIPPath()` is not used anymore and it could be removed.

yfarjoun commented on src/main/java/htsjdk/tribble/TribbleIndexedFeatureReader.java:
```diff
@@ -217,7 +217,7 @@ private void readHeader() throws IOException {
         try {
             is = ParsingUtils.openInputStream(path);
-            if (isGZIPPath(path)) {
+            if (hasBlockCompressedExtension(path) || hasBlockCompressedExtension(new URI(path))) {
```
it seems that the first part of if statement is only good for improving performance, but this isn't a performance bottleneck. is it needed for anything else?

magicDGS replied:
I don't think so, so I will change it. I was following the same approach as in `TribbleIndexedFeatureReader.readHeader()`. I can change it also there or even better, modify the `AbstractFeatureReader.hasBlockCompressedExtension (final File file)` to be something like:

```java
public static boolean hasBlockCompressedExtension (final File file) {
    return hasBlockCompressedExtension(new URI(file));
}
``

magicDGS commented:
I addressed your comment and added other stuff:

* Bgzipped test file (__.bgz__ extension) + test 
* Fixing `WFIterator` (I forgot to do it in the previous commit)
* Deprecation of `isGZIPPath(final String path)`

Because the deprecated method is not used anymore, should I remove it and get rid of the tests too?

Back to you, @yfarjoun!

yfarjoun commented on src/main/java/htsjdk/tribble/TribbleIndexedFeatureReader.java:
```diff
@@ -274,6 +275,7 @@ private void readHeader() throws IOException {
     }
 
     //Visible for testing
+    @Deprecated
```
if deprecating, please comment what should be used instead.

magicDGS replied:
Done, but because it is package specific I wasn't sure if it will be better to remove...

magicDGS commented:
@yfarjoun, I added the comment for the deprecated method.

yfarjoun commented:
💯
